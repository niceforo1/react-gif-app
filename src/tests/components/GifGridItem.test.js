import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas sobre componente Gif Grid Item', () => {
  const url = 'http://localhost/gif.jpg';
  const title = 'Goku vs vegeta';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should be return a GifGridItem rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a <p> with title inside', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should be have an image with alt property equals to title and src equals to url', () => {
    const img = wrapper.find('img');

    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test('div should be have animated__fadeIn class ', () => {
    const div = wrapper.find('div');
    // forma 1
    expect(div.hasClass('animate__fadeIn')).toBe(true);

    // forma 2
    // const className = div.prop('className');
    // expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
