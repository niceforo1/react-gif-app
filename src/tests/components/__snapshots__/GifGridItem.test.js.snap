// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre componente Gif Grid Item should be return a GifGridItem rendered 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Goku vs vegeta"
    src="http://localhost/gif.jpg"
  />
  <p>
    Goku vs vegeta
  </p>
</div>
`;
